database/tbcd/level/headercache.go

```diff
@@ -47,9 +50,31 @@ func (l *lowIQMap) Get(k *chainhash.Hash) (*tbcd.BlockHeader, bool) {
 	return bh, ok
 }
 
-func lowIQMapNew(count int) *lowIQMap {
+func (l *lowIQMap) Purge(k *chainhash.Hash) {
+	l.mtx.RLock()
+	defer l.mtx.RUnlock()
```

Comment:
I think this should call `Lock`/`Unlock`, as it is writing to the map (delete), not reading.
```suggestion
	l.mtx.Lock()
	defer l.mtx.Unlock()
```

Reply:
derp, yep.